Content.Server/NPC/Systems/NpcFactionSelectorSystem.cs

```diff
@@ -0,0 +1,59 @@
+using Content.Server.NPC.Components;
+using Content.Shared.Database;
+using Content.Shared.Popups;
+using Content.Shared.Verbs;
+using Robust.Shared.Prototypes;
+using System.Linq;
+
+namespace Content.Server.NPC.Systems;
+public sealed partial class NpcFactionSelectorSystem : EntitySystem
+{
+
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly IPrototypeManager _prototype = default!;
+    [Dependency] private readonly NpcFactionSystem _factionSystem = default!;
+    [Dependency] private readonly EntityManager _entityManager = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<NpcFactionMemberComponent, GetVerbsEvent<Verb>>(OnGetVerb);
+    }
+
+    private void OnGetVerb(Entity<NpcFactionMemberComponent> entity, ref GetVerbsEvent<Verb> args)
+    {
+        if (!args.CanAccess || !args.CanInteract || args.Hands == null)
+            return;
+
+        NpcFactionSelectorComponent factionSelectorComponent = _entityManager.GetComponent<NpcFactionSelectorComponent>(entity);
+
+        if (factionSelectorComponent.SelectableFactions.Count < 2)
+            return;
```

Comment:
Code style and error safety:
```suggestion
        if (!TryComp<NpcFactionSelectorComponent>(entity, out var factionSelectorComponent) || factionSelectorComponent.SelectableFactions.Count < 2)
            return;
```